### DISCUSSION TITLE
Tenant Contract instead of extending the model?

### DISCUSSION THREAD

**Discussion by inmanturbo** [spatie/laravel-multitenancy#443] [Q&A] [answered]
Is there any interest in a PR which would allow the use of a another tenant model by implementing an interface instead of extending the default? I see this pattern a lot in newer Spatie packages and I'm surprised it's not already been implemented here.

We could also put all of the methods needed into a trait. So any model can use the trait or its own methods to implement the interface. Then we could just get the correct model class from the container wherever it's needed.

This will solve issues caused by  php not supporting multiple inheritance. Thoughts?

**Comment by masterix21** [comment, marked as answer]
Take a look here: 
- https://github.com/spatie/laravel-multitenancy/discussions/543
- https://github.com/spatie/laravel-multitenancy/pull/544

Thanks

**Reply by inmanturbo** [reply]
Hey thanks!
